polaris-react/src/components/MediaCard/MediaCard.tsx

```diff
@@ -66,6 +69,16 @@ export function MediaCard({
     headerMarkup = <div className={styles.Heading}>{headerContent}</div>;
   }
 
+  const dismissButtonMarkup = onDismiss ? (
+    <Button
+      icon={CancelMinor}
+      onClick={() => onDismiss?.()}
```

Comment:
Shouldn't need the optional chain here with the check above
```suggestion
      onClick={onDismiss}
```